src/Database/Table/GroupedSelection.php

```diff
@@ -201,6 +201,10 @@ protected function loadRefCache()
 		}
 	}
 
+	protected function emptyResultSet($saveCache = TRUE, $deleteRererencedCache = TRUE) {
+		parent::emptyResultSet($saveCache, false);
```

Comment:
I don't see any difference between your code and this solution:

```
protected function GroupedSelection::emptyResultSet(..) {
  parent::emptyResultSet($saveCache);
  $this->refCache['referenced'] = [];
}
```

Where is a difference?